src/status_im/contexts/profile/edit/accent_colour/view.cljs

```diff
@@ -0,0 +1,61 @@
+(ns status-im.contexts.profile.edit.accent-colour.view
+  (:require [quo.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [status-im.constants :as constants]
+            [status-im.contexts.profile.edit.accent-colour.style :as style]
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [utils.responsiveness :as responsiveness]))
+
+(defn view
+  []
+  (let [insets                (safe-area/get-insets)
+        {window-width :width} (rn/get-window)
+        profile               (rf/sub [:profile/profile-with-image])
+        customization-color   (rf/sub [:profile/customization-color])
+        profile-picture       (profile.utils/photo profile)
+        display-name          (profile.utils/displayed-name profile)
+        custom-color          (reagent/atom (or customization-color
+                                                constants/profile-default-color))
+        on-change             #(reset! custom-color %)]
+    (fn []
+      [quo/overlay
+       {:type            :shell
+        :container-style (style/page-wrapper insets)}
+       [quo/page-nav
+        {:key        :header
+         :background :blur
+         :icon-name  :i/arrow-left
+         :on-press   #(rf/dispatch [:navigate-back])}]
+       [rn/view
+        {:key   :content
+         :style style/screen-container}
+        [rn/view {:style {:flex 1}}
+         [quo/text-combinations {:title (i18n/label :t/accent-colour)}]
+         [quo/profile-card
+          {:profile-picture     profile-picture
+           :name                display-name
+           :card-style          style/profile-card
+           :customization-color @custom-color}]
+         [quo/text
+          {:size   :paragraph-2
+           :weight :medium
+           :style  style/color-title}
+          (i18n/label :t/accent-colour)]
+         [quo/color-picker
+          {:blur?            true
+           :default-selected @custom-color
+           :on-change        on-change
+           :window-width     window-width
+           :container-style  {:padding-left (responsiveness/iphone-11-Pro-20-pixel-from-width
```

Comment:
It was strange for me too, i implemented like other usage in the code. will search through git history if i can find any reason for this exception

Reply:
I think this function is meh... The views shouldn't worry about such details directly.

My take on this would be:

- Rename to `responsiveness/ratio-to-viewport`
- Inside `ratio-to-viewport`, check which device is in use before doing any calculation. Don't make public device-specific functions in that namespace, otherwise devs will more likely use them directly. Therefore, we would only expose a general responsiveness API.

Reply:
this function was introduced here https://github.com/status-im/status-mobile/pull/17748#discussion_r1377381914, maybe i should rename func to something that shows it's related to design pixel match ?

Reply:
in edit accent color there is no left padding, i removed it

Reply:
@mohsen-ghafouri, running in the risk of repeating myself even if you understood me, sorry about that :) My point here was that the view shouldn't call a device specific function in the view. There are too many devices, this should be abstracted in a namespace exposing a public function with a more general name and this function should switch/detect which device it's running, otherwise it would force the caller (view) to do this detection (or people will forget, as it happened here).

But we don't need to do what I'm saying, seems overkill for the moment.

> It was strange for me too, i implemented like other usage in the code.

Yep, we have a huge team, and the quality of the code is questionable in many areas, but they still get merged.

Thank you for the patience in this thread

Reply:
I understand @ilmotta, and thanks for insightful explanation. yes perfection is hard to achieve in a large team.